workflow/controller/exec_control.go

```diff
@@ -72,20 +72,25 @@ func (woc *wfOperationCtx) applyExecutionControl(pod *apiv1.Pod, wfNodesLock *sy
 			woc.log.Warnf("Failed to unmarshal execution control from pod %s", pod.Name)
 		}
 	}
+	containerName := common.WaitContainerName
+	// ResourceTemplate pod will have main container only
```

Comment:
minor - I think this comment could be clearer - if I understand correctly

`a resource template does not have a wait container, instead the only container is the main container (which is running argoexec)`